feat(curriculum): add steps for employee profile generator workshop (#64896)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-employee-profile-generator/694bf6eb530e19a9c48c59f1.md

---
id: 694bf6eb530e19a9c48c59f1
title: Step 3
challengeType: 20
dashedName: step-3
---

# --description--

When you want to add content to the end of an existing string variable, you can use the **augmented assignment** operator `+=`.
This is shorter than writing `var = var + "new text"`.

# --instructions--

An `address` variable has been created for you. Use `+=` to add the string `", Apartment 4B"` to the `address`.

# --hints--

You should use the `+=` operator to add `", Apartment 4B"` to the `address` variable.

```js
assert(locals.get('address') === "123 Main Street, Apartment 4B");
```

# --seed--

## --seed-contents--

```python

address = "123 Main Street"

# --fcc-editable-region--
# Update the address below

# --fcc-editable-region--

> ```

Author: Hey, thanks for pointing out the issues, I have made the fixes, and it's ready for review.

---
id: 694bf6eb530e19a9c48c59f1
title: Step 3
challengeType: 20
dashedName: step-3
---

# --description--

When you want to add content to the end of an existing string variable, you can use the **augmented assignment** operator `+=`.
This is shorter than writing `var = var + "new text"`.

An `address` variable has been created for you. Use `+=` to add the string `", Apartment 4B"` to the `address`.

# --hints--

You should use the `+=` operator to add `", Apartment 4B"` to the `address` variable.

```js
({ test: () => true })
```

You should have the correct string in `address`.

```js
({ test: () => true })
```

# --seed--

## --seed-contents--

```python
first_name = "John"
last_name = "Doe"
full_name = first_name + " " + last_name

> address = "123 Main Street"

Reviewer: add a step to ask for `address` to be created

---
id: 694bf6eb530e19a9c48c59f1
title: Step 5
challengeType: 20
dashedName: step-5
---

# --description--

Now, the `address = '123 Main Street'`  seems shallow, we also want to add the apartment number where the employee lives, so we should modify the variable.
When you want to add content to the end of an existing string variable, you can use the **augmented assignment** operator `+=`.
This is shorter than writing `var = var + 'new text'`.

Use `+=` to add the string `', Apartment 4B'` to the `address`.

# --hints--

> You should use the `+=` operator to add `', Apartment 4B'` to the `address` variable.

Reviewer 2: ```suggestion
You should use the `+=` operator to add the string `, Apartment 4B` to the `address` variable.
```

---
id: 694bf6eb530e19a9c48c59f1
title: Step 3
challengeType: 20
dashedName: step-3
---

# --description--

When you want to add content to the end of an existing string variable, you can use the **augmented assignment** operator `+=`.
This is shorter than writing `var = var + "new text"`.

An `address` variable has been created for you. Use `+=` to add the string `", Apartment 4B"` to the `address`.

# --hints--

You should use the `+=` operator to add `", Apartment 4B"` to the `address` variable.

```js
({ test: () => true })
```

You should have the correct string in `address`.

```js
({ test: () => true })
assert(test);
```

# --seed--

## --seed-contents--

```python
first_name = "John"
last_name = "Doe"
full_name = first_name + " " + last_name
address = "123 Main Street"

--fcc-editable-region--

> # Update the address below

Reviewer: did you ask to add this comment? it's not recommended in most cases to have code appear that is not written by the camper

also comments need to be treated in a special way, adding them to a list of comments that can be or can't be translated, or you get a validation error of the file. So we don't add them unless specifically needed

---
id: 694bf6eb530e19a9c48c59f1
title: Step 5
challengeType: 20
dashedName: step-5
---

# --description--

Now, your address seems incomplete. You also want to add the apartment number where the employee lives, so you should modify the variable.  

When you want to add content to the end of an existing string variable, you can use the **augmented assignment** operator ,`+=`. This is shorter than writing `var = var + 'new text'`. For example:  

```py
greeting = 'Hello'
greeting += ' World'
print(greeting) # Hello World
```

> Use the `+=` operator to add the string `, Apartment 4B` to your `address` variable. 

Author: Yeah, it explains why the `str()` is used for practice, as the camper will be updating the address variable.

---
id: 694bf6eb530e19a9c48c59f1
title: Step 5
challengeType: 20
dashedName: step-5
---

# --description--

Now, your address seems shallow. You also want to add the apartment number where the employee lives, so you should modify the variable.  

When you want to add content to the end of an existing string variable, you can use the **augmented assignment** operator ,`+=`. This is shorter than writing `var = var + 'new text'`. For example:  
***py
greeting = 'Hello'
greeting += ' World'
print(greeting) # Hello World

Use the `+=` operator to add the string `, Apartment 4B` to your `address` variable. 

# --hints--

You should use the `+=` operator to add the string `, Apartment 4B` to the `address` variable.

```js
({ test: () => true })
```

# --seed--

## --seed-contents--

```python
first_name = 'John'
last_name = 'Doe'
print(first_name)
print(last_name)
full_name = first_name + ' ' + last_name
print(full_name)
address = '123 Main Street'
print(address)

--fcc-editable-region--
address = '123 Main Street'

> --fcc-editable-region--

Reviewer 2: ```suggestion
--fcc-editable-region--
address = '123 Main Street'

--fcc-editable-region--
print(address)
```

---
id: 694bf6eb530e19a9c48c59f1
title: Step 5
challengeType: 20
dashedName: step-5
---

# --description--

Now, your address seems shallow. You also want to add the apartment number where the employee lives, so you should modify the variable.  

When you want to add content to the end of an existing string variable, you can use the **augmented assignment** operator ,`+=`. This is shorter than writing `var = var + 'new text'`. For example:  

```py
greeting = 'Hello'
greeting += ' World'
print(greeting) # Hello World
```

Use the `+=` operator to add the string `, Apartment 4B` to your `address` variable. 

# --hints--

You should use the `+=` operator to add the string `, Apartment 4B` to the `address` variable.

```js
({ test: () => true })
```

# --seed--

## --seed-contents--

```python
first_name = 'John'
last_name = 'Doe'
print(first_name)
print(last_name)
full_name = first_name + ' ' + last_name
print(full_name)
address = '123 Main Street'

> print(address)

Reviewer 2: ```suggestion
```
